junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -54,6 +54,9 @@ private Assertions() {
 	 * <p>See Javadoc for {@link #fail(String, Throwable)} for an explanation of
 	 * this method's generic return type {@code V}.
 	 */
+	// we purposely return a V that has nothing to do with the parameters of this method; see
+	// the javadoc for details.
+	@SuppressWarnings("TypeParameterUnusedInFormals")
```

Comment:
I'm simply hesitant to introduce all of these `@SuppressWarnings` statements for non-standard warning labels since that will result in warnings about unknown warning labels in IDEs... which will annoy the hell out of me.

Is there not a more elegant way to achieve this without introducing `@SuppressWarnings` statements that use proprietary warning labels?